File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");
    int wordcount = 1;
    int lettercount = 0;
    int length = strlen(text);
    int n;
    for (n = 0; n < length; n++)
    {
        if (isalpha(text[n]))
        {
            lettercount += 1;
        }
        else if (text[n] == ' ')
        {
            wordcount += 1;
        }
    }
    float L = lettercount / ((float) wordcount / 100);
    int fullstopcount = 0;
    for (n = 0; n < length; n++)
    {
        if (text[n] == '.')
        {
            fullstopcount += 1;
        }
        if (text[n] == '?')
        {
            fullstopcount += 1;
        }
        if (text[n] == '!')
        {
            fullstopcount += 1;
        }
    }
    float S = fullstopcount / ((float) wordcount / 100);
    float index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) index);
    }
}

File: readability.py
text = input("Text: ")
wordcount = 1
lettercount = 0
length = len(text)

for n in range(length):
    if text[n].isalpha():
        lettercount += 1
    elif text[n] == " ":
        wordcount += 1

l = lettercount / (float(wordcount)/100)
fullstopcount = 0

for n in range(length):
    if text[n] == '.' or text[n] == '?' or text[n] == '!':
        fullstopcount += 1

s = fullstopcount / (float(wordcount) / 100)
index = 0.0588 * l - 0.296 * s - 15.8
index = round(index)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

